### PR TITLE
Feature/51974 ps report logging

### PR DESCRIPTION
**Pull request by GuyHarwood**
create `PsReportLogger` that combines stdio logging via `context.log` and messages that go onto a logging queue.
These are then processed into text files persisted in blob storage.

### CODE REVIEW

File: tslib/src/functions-ps-report/ps-report-5-log-writer/index.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      messageCount += messageBatch.length
+      messageBatch = await receiver.receiveMessages(config.PsReportLogWriter.MessagesPerBatch)
+    }
+    context.log(`${functionName}: processed ${messageCount} messages...`)
+    await disconnect()
+    finish(start, context)
+    return
+  } catch (error) {
+    context.log.error(error)
+    /*     if (!RA.isNilOrEmpty(messageBatch)) {
```
Comment (jon-shipley) [inline tslib/src/functions-ps-report/ps-report-5-log-writer/index.ts:64]: Should this be uncommented or deleted?

File: tslib/src/functions-ps-report/ps-report-5-log-writer/log.service.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  private readonly writer: IPsLogWriter
+  private readonly parser: PsLogMessageParser
+
+  constructor (logWriter?: IPsLogWriter) {
+    this.generator = new PsLogSetGeneratorService()
+    this.parser = new PsLogMessageParser()
+    this.writer = logWriter ?? new PsLogWriter()
+  }
+
+  async createV2 (setId: string, messages: IServiceBusMessageLike[]): Promise<void> {
```
[the remark this answers is not in the document]
Comment (GuyHarwood) [reply tslib/src/functions-ps-report/ps-report-5-log-writer/log.service.ts]: this was an interim from an old version. thanks for catching it

File: func-ps-report/ps-report-5-log-writer/function.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "bindings": [
+    {
+      "name": "psReport5LogWriter",
+      "type": "timerTrigger",
+      "direction": "in",
+      "timerDescription": "{second} {minute} {hour} {day} {month} {day-of-week}",
+      "dev-schedule": "0 */5 * * * *",
+      "prod-schedule": "0 0 6 * * 3",
+      "prod-description": "Every Wednesday at 6am",
+      "schedule": "0 0 22 * * 2"
```
[the remark this answers is not in the document]
Comment (GuyHarwood) [reply func-ps-report/ps-report-5-log-writer/function.json]: thanks for reminder!

**Review by jon-shipley — APPROVED** [review]
LGTM - many thanks for the changes 💯

**Pull request merged**